### PR TITLE
Whisper/scottg notebook

### PR DESCRIPTION
**Pull request by scott-griffiths**
Rewriting Whisper inference notebook.

Adding explanatory text, refactoring code, adding multimedia features and updating for new optimum-graphcore version.
Tested with optimum-graphcore 0.6.0.

### CODE REVIEW

File: notebooks/whisper-example.ipynb [hunk 1]
```diff
@@ -1,354 +1,351 @@
...
-   },
-   "outputs": [],
+   "cell_type": "markdown",
+   "id": "aa59411d",
+   "metadata": {},
    "source": [
-    "# Choose here the index of the test example to use\n",
-    "test_idx = 4"
+    "In the next cell we compare the expected text from the dataset with the transcribed text.\n",
+    "There will typically be some small differences, but even the 'tiny' Whisper model does a great job, and also adds in punctuation.\n",
```
Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:182]:
```suggestion
    "There will typically be some small differences, but even 'whisper-tiny' does a great job! It even adds punctuation.\n",
```

File: notebooks/whisper-example.ipynb [hunk 2]
```diff
@@ -1,354 +1,351 @@
...
+    "\n",
+    "You can compare with the audio itself using the controls underneath."
```
Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:184]:
```suggestion
    "You can listen to the audio and compare the model result yourself using the controls below."
```

File: notebooks/whisper-example.ipynb [hunk 3]
```diff
@@ -1,354 +1,351 @@
...
...0qALsAXwLwAR8D4QPuAnUFHAYGB5MJMQjIBp0IjQcrCXMI+AcBC9sICgf/Bj8KPQmkCVkHLAlrBUcEYQZCBtcH8Qf5Ay0D5wXjAngGwgHtAnUCAATW/p0EGP/sAV//5AKTAi0AGAJFAMT9/P3C/6/91wB5ALf9Pv9m/Sr8jf63/lr9hP+4/F/9X/oy/pz9lvw8/Ub+PP3r/8T7H/oR/FD8Af1M/Mz+Ev4v/iH5L/6g+kv+mPsh/Rv+If2o/i//W/yG/d3+ev00/3v/e/xq/2j9ywLU/WsALv09ApX/VgGPARv+MgMnAawBcQDBAu7/UgPgAWIC1gOW/8gDLwS/AhIDxgOWAZcDAwRXAsoEigSZ/bUD/QH3A+IAewfl+hkEkv45BFYB9gCvBB4BuQCaAdj+2P7JAmr/MQGAAGwBaf1D+2wBxf8TAfgAz/2b++0A5vxQAlX9NP/2/scBDPkwAM/+xP9y/v3/hgAd/5L+P/mC/9j/8vwIBPT7Jv2q/g8AYQB8/fUB0vlOAT77KAD1/PEDCfzzBF/5JgD2+8T8MQI8APYDef7EAdH7FAFT/TQA2wFyBJQB2v//Aiv+eQH0AFT+5AWuAXT++QSK/twDmgE0AKUBtgHSAAf9NwcyAEkDFgTm/DEEKAD8/EMEFQZL/aP6bws3+pj53Apu/779dwOsAIYDcP0kBh/5ogK0A/IAafjrAQAE+v0RAZMB5AIy+rgBvvlb/5YB9QLs/jf9IQao+lL9BQRw/F8DF/42/8r7pQEiAbICS/4Z/oT8aQGYAYoETv0rAK77y/9a/a8EVf0yBJL2HwaX+QcC2gGB/QUBZv0DB3P98f87/EQG/P2e/ZYAW/+JAQgBj/94Arz9oQCP/58BwQPV/0X+DP/EAfX/LwGyA2cEgvy1/3P9xP8R/xwDUQXl/qkGvPou/Hf+8AInAUsCmv/xAWv85AbI+esC4P7R/soA5QDXAlYBQABJ/Hz5qQKgBOP5DQfgARQD//ux/2z7ZAPkBLH++PzSA8788/7++oEGbAJy/hYEgfpvAI36XQHe/oD/Kvw0AQ4Flv8HAuf7yv7C+2ECZv7KAyoBzvsIBGf4agFHBOv8c/8A+7YAzAGa/zP/ZPkQA0oAzf5SAJICjv2I/h7+pf+g/oUAzgFXANME+vsEAgD76wG2/iIFcfwtB9T3sP1qA4MADQDfAgL4Nw
```
Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:255]:
```suggestion
    "The model only needs to be compiled once. Subsequent inferences will be much faster.\n",
```

Comment (jayniep-gc) [reply notebooks/whisper-example.ipynb:255]: I'm not sure why GitHub marks the image for deletion. 😞

Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:320]:
```suggestion
    "## Next steps\n",
```

Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:322]:
```suggestion
    "The `whisper-tiny` model used here is very fast for inference and cheap to run, but its accuracy can be improved.\n",
```

Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:323]:
```suggestion
    "The `whisper-base` and `whisper-small` models have 74M and 244M parameters respectively (compared to just 39M for `whisper-tiny`). You can try out `whisper-base` by using `model_spec = \"openai/whisper-base.en\"` or `whisper-small` by using `model_spec = \"openai/whisper-small.en\"` near the start of this notebook.\n",
```

Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:329]: Add a link for the contact Graphcore. Other notebooks refer to the Graphcore Slack Community so maybe that would be suitable here.

```suggestion
    "To run these you will need more than the IPU-POD4. On Paperspace this is available using either an IPU-POD16 or a Bow Pod16 machine. Please contact Graphcore if you need assistance running these larger models.\n"
```

Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:341]:
```suggestion
    "We used the Optimum Graphcore package to interface between the IPU and the Hugging Face Transformers library. This meant that only a few lines of code were needed to get this state-of-the-art automated speech recognition model running on IPUs."
```

**Review by jayniep-gc — CHANGES_REQUESTED** [review]
I've made my changes with the main aim of being clearer for the user. Please let me know if anything in what I've done is not clear. 

There is a problem with GitHub. When I try to add comments to anything that appears after a figure, an image is marked for replacement even though this is not intended. The last few comments have this and the changed text is not highlighted...

File: notebooks/whisper-example.ipynb [hunk 4]
```diff
@@ -1,354 +1,351 @@
...
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "b458fecc",
+   "metadata": {},
    "source": [
-    "%%capture\n",
-    "%pip install git+https://github.com/graphcore/optimum-graphcore-fork.git@whisper/poc \"tokenizers<0.13\""
+    "## Environment setup\n",
+    "\n",
+    "In order to run this notebook you will need to be in an environment with the Poplar SDK installed and enabled - on Paperspace this is handled by default.\n",
```
Comment (jayniep-gc) [inline notebooks/whisper-example.ipynb:30]:
```suggestion
    "In order to run this notebook you will need to be in an environment with the Poplar SDK installed and enabled. This is done by default on Paperspace. If you are not using Paperspace, refer to the [getting started guide](https://docs.graphcore.ai/en/latest/getting-started.html#getting-started) for your system for a description of how to set this up.\n",
```

**Review by jayniep-gc — COMMENTED** [review]
Phrasing improvement to conclusion

**Review by jayniep-gc — COMMENTED** [review]
Fixing Join Slack link

**Comment by payoto** [comment]
Closed in favour of an upstream PR https://github.com/huggingface/optimum-graphcore/pull/312

**Pull request closed without merging**